syntactic/src/test/scala/syntactic_checker/whitelist/WhitelistCheckerTestRunner.scala

```diff
@@ -0,0 +1,186 @@
+   * @param dialect               the dialect to be used by the checker
+   * @param excludedFeatures      the checker will use all atomic features except these ones
+   * @param expectedViolationCnts map from line indices to the number of expected violations on the
+   *                              corresponding line. Lines with 0 expected violation can be omitted
+   */
+  def expectInvalidWhenExcludingFeatures(
+                                          srcFileName: String,
+                                          dialect: Dialect = defaultDialect,
+                                          excludedFeatures: List[Feature],
+                                          expectedViolationCnts: Map[Int, Int]
+                                        ): Unit = {
+    val allowedFeatures = allFeatures.filter(!excludedFeatures.contains(_))
+    runTest(srcFileName, expectInvalidCheckFunc(expectedViolationCnts), dialect, allowedFeatures)
+  }
+
+  /**
+   * Test method for a test where the parsing of the program should fail <p>
+   * The method therefore expects a syntactic_checker.CheckResult.ParsingError to be returned by the check method
+   *
+   * @param srcFileName name of the file containing the program <p>
+   *                    This is the name of one of the files in test/res, without ".scala"
+   * @param dialect     the dialect to be used by the checker
+   */
+  def expectParsingError(
+                          srcFileName: String,
+                          dialect: Dialect = defaultDialect
+                        ): Unit = {
+    runTest(srcFileName, expectParsingErrorCheckFunc, dialect, allFeatures)
+  }
+
+  private val defaultDialect = dialects.Scala3
+  private val testResourcesDirectory = "src/test/res"
+  private val testFilesExtension = "scala"
+
+  /**
+   * Runs the test described by its arguments
+   *
+   * @param srcFileName   name of the file containing the program <p>
+   *                      This is the name of one of the files in test/res, without ".scala"
+   * @param assertionFunc the function that is called on the syntactic_checker.CheckResult returned by the check method <p>
+   *                      This function should contain the assertions
+   * @param dialect       the dialect to be used by the checker
+   * @param features      the features that the checker should use
+   */
+  private def runTest(
+                       srcFileName: String,
+                       assertionFunc: CheckResult[WhitelistViolation] => Unit,
+                       dialect: Dialect,
+                       features: List[Feature]
+                     ): Unit = {
+    val checker = WhitelistChecker(features)
+    // FIXME should probably not be necessary to have both paths
+    val intellijTestFile = AbsolutePath(s"$testResourcesDirectory/$srcFileName.$testFilesExtension").toFile
+
+    def sbtTestFile = AbsolutePath(s"syntactic/$testResourcesDirectory/$srcFileName.$testFilesExtension").toFile
```

Comment:
I would argue this is a bug in IntelliJ, it seems that it sets the working directory to be the module root directory, whereas sbt sets the working directory to be the project root directory.

Could you maybe configure IntelliJ to use the “correct” working directory?

Reply:
An alternative would be to use “test resources” instead of files. Those would be equally loaded by IntelliJ and sbt.

Reply:
By default, the directory for test resources is `src/test/resources`. You can load resources with `Source.fromResource` for instance.

Reply:
I managed to configure Intellij to use the same directory as sbt, thanks!
Fixed in 7a4dca952b16dad96c364fca76e8c72251a0e7ec